### PR TITLE
5392 members table

### PR DESCRIPTION
**Pull request by lzach83**
## Link to Issue
Closes: #5392

## Description of Changes
- Added more verbose filters
- Added group sorting inline 
- Sorted group tags alphabetically 

## "How We Fixed It"

Sorted the group names after the members get sorted.

## Test Plan
- Make multiple members that have joined groups and ensure they follow the DOD

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/Community/Members/MembersSection/MembersSection.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Members/MembersSection/MembersSection.tsx:48]: i think we should do this in the `CommunityMembersPage`

File: packages/commonwealth/client/scripts/views/pages/Community/Members/GroupsSection/GroupsSection.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Members/GroupsSection/GroupsSection.tsx:76]: nice spot here, i think we should also check for null `app?.user?.activeAccount` -- it breaks if not login I think

File: packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx:129]: hmm, so i actually did `"All groups" -> "All"` since it was also used in the members tab. 

I think we should revert this to "All" or maybe use both "All groups" / "All members" depending on the active tab

**Review by mzparacha — CHANGES_REQUESTED** [review]
Small changes. Thanks for the pr

File: packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx
Comment (lzach83) [reply packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx:129]: Was just following the [figma](https://www.figma.com/file/ePuGFatYQI6GL72i27sVwj/Gating?node-id=1221%3A31747&mode=dev)

Which is correct? @zakhap

File: packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx
Comment (lzach83) [reply packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx:113]: This was just because they were built off each other, since group uniqueness was merged -- i'll sync with master and clear this.

File: packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx:143]: since .sort will sort in-place/reference-update -- this will effect the react query cache, so I suggest we clone the `members` above i.e `[...members.pages]`

File: packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx:229]: same as above

File: packages/commonwealth/client/scripts/views/pages/Community/Members/GroupsSection/GroupsSection.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Members/GroupsSection/GroupsSection.tsx:79]:
```diff
-- isJoined={(group.members || []).find((x) => {
--                if (!app.user.activeAccount || app.user.activeAccount === null)
--                 return;
--               x?.address?.address === app.user.activeAccount.address;
--            })}
++ isJoined={(group.members || []).find(
++                (x) => x?.address?.address === app?.user?.activeAccount?.address
++              )}
```

File: packages/commonwealth/client/scripts/views/pages/Community/Members/index.types.ts
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/Community/Members/index.types.ts:1]: again I think we should do 'all' instead of 'all groups' or change labels when we change tabs

File: packages/commonwealth/client/scripts/views/pages/view_thread/ViewThreadPage.tsx
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/view_thread/ViewThreadPage.tsx:1]: not reviewing this file as it got updated in https://github.com/hicommonwealth/commonwealth/issues/5397 (I did some updates there)

**Comment by mzparacha** [comment]
The group section "in group" / "not in group" tags are broken.

https://github.com/hicommonwealth/commonwealth/assets/51641047/9c55a3c9-8dbc-4e5e-8103-4802cb7d0763

**Review by mzparacha — CHANGES_REQUESTED** [review]
Thanks for the updates, still need some changes.

File: packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx
Comment (zakhap) [reply packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx:133]: Let's stick the Figma!

**Comment by lzach83** [comment]
> The group section "in group" / "not in group" tags are broken. (tested on master, works well there)
> 
>  Screen.Recording.2023-11-01.at.10.45.22.AM.mov

All because i forgot a return. Fixed in most recent commit.

File: packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx:229]: just noticed this is expected

**Review by mzparacha — APPROVED** [review]
LGTM

**Review by rbennettcw — APPROVED** [review]

File: packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx
Comment (lzach83) [reply packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx]: @mzparacha @rbennettcw -- i know i have two approvals but can y'all re-check this with my newest commit https://github.com/hicommonwealth/commonwealth/pull/5512/commits/211a98af57d2bc4e0476af82b72edb11c87a32a8

**Comment by zakhap** [comment]
LGTM too!

**Review by mzparacha — APPROVED** [review]
LGTM after [latest changes](https://github.com/hicommonwealth/commonwealth/pull/5512#discussion_r1380282234)

File: packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Members/CommunityMembersPage.tsx]: Resolving, thanks

**Pull request merged**
